clients/bigquery/merge.go

```diff
@@ -42,17 +43,27 @@ func merge(tableData *optimization.TableData) (string, error) {
 			colKind := tableData.InMemoryColumns[col]
 			colVal := value[col]
 			if colVal != nil {
-				switch colKind {
-				case typing.DateTime:
-					ts, err := typing.ParseDateTime(fmt.Sprint(colVal))
-					if err != nil {
-						return "", fmt.Errorf("failed to cast colVal as time.Time, colVal: %v, err: %v", colVal, err)
+				switch colKind.Kind {
+				case typing.ETime.Kind:
+					eTime, isOk := colVal.(*typing.ExtendedTime)
+					if !isOk {
+						var err error
+						eTime, err = typing.ParseExtendedDateTime(fmt.Sprint(colVal))
+						if err != nil {
+							return "", fmt.Errorf("failed to cast colVal as time.Time, colVal: %v, err: %v", colVal, err)
+						}
 					}
 
-					// We need to re-cast the timestamp INTO ISO-8601.
-					colVal = fmt.Sprintf("PARSE_DATETIME('%s', '%v')", RFC3339Format, ts.Format(time.RFC3339Nano))
+					switch eTime.NestedKind.Type {
+					case typing.DateTimeKindType:
```

Comment:
We need to override this format for now because BigQuery requires us to rewrite all of our formats to their specification. We will add this when a customer asks for it.